src/univariate/continuous/normal.jl

```diff
@@ -1,12 +1,18 @@
-immutable Normal <: ContinuousUnivariateDistribution
-    μ::Float64
-    σ::Float64
+immutable Normal{T<:Number} <: ContinuousUnivariateDistribution
+    μ::T
+    σ::T
 
-    Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); new(μ, σ))
-    Normal(μ::Real) = Normal(μ, 1.0)
-    Normal() = Normal(0.0, 1.0)
+    Normal(μ::T, σ::T) = new(μ, σ)
 end
 
+Normal{T<:Number}(μ::T, σ::T) = Normal{T}(μ, σ)
+Normal(μ::Number, σ::Number) = Normal(promote(μ, σ)...)
+Normal(μ::Number) = Normal(μ, one(μ))
+#Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); Normal(μ, σ))
+Normal(μ::Real) = Normal(μ, one(μ))
+Normal() = Normal(0.0, 1.0)
+Normal(μ::Complex, σ::Complex) = error("Normal with complex mean and variance not defined")
```

Comment:
I remember when I first coded `DualNumber` in par (in terms of style) with `Complex` 2 years ago, there was a brief discussion about adding them to Base. This didn't happen at that time, but autodiff has come a rather immediate (if not long) way in Julia since then. While Julia has been developing, there have been several and more central breakages than that of `HyperReal`. I think the central point of discussion is how vital and central as a concept we consider automatic differentiation (and thus associated numbers) to be. If we think that in fact AD is a rather important tool for derivative computations in scientific computing (such as optimizations and MCMC for ex),despite the fact that AD has perhaps not received the magnitude of attention it deserves in the wider scientific community, then we could seize this opportunity and give it a more glorious position as an innovative step. If on the other hand the consensus is that AD is not as crucial so as to promote derivative computations at a different scale, then yes, it is not an optimal choice to add an abstract HyperReal type in Base. Perhaps we could go with your solution, despite my view of how underrated AD is...